chore(docs): deprecation notices for methods and properties

petebacondarwin opened the pull request:
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**



**What is the current behavior? (You can also link to an open issue here)**



**What is the new behavior (if this is a feature change)?**



**Does this PR introduce a breaking change?**



**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:


Closes #15351

petebacondarwin commented:
I'm aware that Travis is not happy. I will fix this later today and maybe even break the changes into smaller commits for easier reviewing :-)

petebacondarwin committed:
chore(package.json): update to latest dgeni-packages

petebacondarwin committed:
chore(docs): deprecation notices for methods and properties

Closes #15351

petebacondarwin commented:
OK, smaller commits. Travis is happy.
What more could you want of a PR?

mgol commented on src/ngCookies/cookieStore.js:
```diff
@@ -5,6 +5,8 @@ angular.module('ngCookies').
  * @ngdoc service
  * @name $cookieStore
  * @deprecated
+ * Please use the {@link ngCookies.$cookies `$cookies`} service instead.
```
Hasn't this been deprecated for some time now? Perhaps we should schedule a removal timeline?

mgol replied:
It'd be also nice to mention the version in which this was deprecated.

mgol replied:
Sure. :)

Narretz commented on docs/app/assets/css/docs.css:
```diff
@@ -662,6 +662,12 @@ ul.events > li {
   max-width: 100%;
 }
 
+.deprecated-title {
```
Shouldn't this technically be part of the deprecation commit?

Narretz commented:
Good stuff. Having the templates in the repo allows us to move faste with changes, too.
A few notes:
-  I think using "warning" is a bit too weak for deprecation. It should be danger.
- The API deprecation template is different. We should ideally use the same as for props and methods (can do in a follow-up)
- In a follow-up, we should also make the deprecation notices in `@description` blocks consistent. Most noticable in the compile docs

mgol commented:
> Can we have "sub-annoations" for `@deprecated`? then we could insert `@since_version`, `@remove_version`, and `@description`, and make this consistent across the board

That would be wonderful.

gkalpak commented on docs/config/templates/ngdoc/api/module.template.html:
```diff
@@ -5,41 +5,33 @@
+
+  {% code %}
+      <script src="angular.js">
+      <script src="{$ doc.packageFile $}">
```
Missing closing `<script>` tags.

gkalpak commented on docs/config/templates/ngdoc/api/module.template.html:
```diff
@@ -5,41 +5,33 @@
-      <a href="http://bower.io">Bower</a> e.g.
-      {% code %}bower install {$ doc.packageName $}#X.Y.Z{% endcode %}
+      <a href="http://bower.io">Bower</a><br>
+      e.g. {% code %}bower install {$ doc.packageName $}@X.Y.Z{% endcode %}
```
Shouldn't this be `#` instead of `@`?

gkalpak commented on docs/config/templates/ngdoc/api/module.template.html:
```diff
@@ -69,7 +61,6 @@ <h2 id="known-issues">Known Issues</h2>
 {% endif %}
 
 
-{% if doc.componentGroups.length %}
```
Why was this removed?

petebacondarwin replied:
this is what is in dgeni-packages - so it looks like a breakage - but then it is re-fixed in a subsequent commit.

petebacondarwin commented:
> Can we have "sub-annoations" for @deprecated? then we could insert @since_version, @remove_version, and @description, and make this consistent across the board

The content of an annotation is completely configurable. But it can't really use sub-annotations, because the parser would see them as no top level annotations.
But you can do something like:

```
@deprecated since-version="..." remove-version="..." what is left is description
```

Then we would just parse for the attributes at the start of the description and pull them out.

Is that what you want?

petebacondarwin commented:
Fixed up.

Narretz commented:
@petebacondarwin  Yes, this custom annoation content like you describe is what I'd like. I think `@param` does something like that? It has some custom `transforms` in the tag-defs. I was lead a bit in a wrong direction by looking a `@method` which seems to have "real" sub-annoations, but then again that's jsdoc, which is different.

petebacondarwin committed:
chore(docs): bring in all templates from dgeni-packages

petebacondarwin commented:
OK, so I have updated the `@deprecation` tag to take `sinceVersion` and `removeVersion` options.

I also moved the docs.css file changes to the correct commit :-)

@Narretz PTAL

mgol commented:
There are some linting violations but it's most likely due to us not checking docs properly. We can fix it later.

The new `@deprecated` syntax is not yet used, is it, @petebacondarwin?

mgol commented:
The deprecation notice has disappeared from the `ngComponentRouter` page.

petebacondarwin committed:
docs(*): fix up deprecation notices
